packages/fuels-core/src/types/param_types/decodable_debug.rs

```diff
@@ -0,0 +1,241 @@
+use std::iter::zip;
+
+use crate::{
+    codec::ABIDecoder,
+    types::{
+        errors::{error, Result},
+        param_types::ParamType,
+        Token,
+    },
+};
+
+impl ParamType {
+    pub fn decode_debug(&self, bytes: &[u8]) -> Result<String> {
```

Comment:
I'd move this to the decoder for two reasons:
1. It is not expected of the ParamType to decode but rather to describe a type
2. If moved to the decoder then you wouldn't have to make the decision to use the default decoder -- i.e. the user chooses the decoder. 

So something like
`decoder.decode_as_debug_str(param_type, bytes)?`